src/pages/ReportDetailsPage.tsx

```diff
@@ -1,103 +1,103 @@
-import {withNetwork} from '@components/OnyxProvider';
 import ParentNavigationSubtitle from '@components/ParentNavigationSubtitle';
-import participantPropTypes from '@components/participantPropTypes';
 import PressableWithoutFeedback from '@components/Pressable/PressableWithoutFeedback';
 import RoomHeaderAvatars from '@components/RoomHeaderAvatars';
 import ScreenWrapper from '@components/ScreenWrapper';
-import withLocalize, {withLocalizePropTypes} from '@components/withLocalize';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
 import useThemeStyles from '@hooks/useThemeStyles';
-import compose from '@libs/compose';
 import Navigation from '@libs/Navigation/Navigation';
 import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as PolicyUtils from '@libs/PolicyUtils';
 import * as ReportUtils from '@libs/ReportUtils';
 import * as Report from '@userActions/Report';
 import CONST from '@src/CONST';
+import type {TranslationPaths} from '@src/languages/types';
 import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
+import type * as OnyxTypes from '@src/types/onyx';
+import type DeepValueOf from '@src/types/utils/DeepValueOf';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import type {WithReportOrNotFoundProps} from './home/report/withReportOrNotFound';
 import withReportOrNotFound from './home/report/withReportOrNotFound';
-import reportPropTypes from './reportPropTypes';
 
-const propTypes = {
-    ...withLocalizePropTypes,
-
-    /** The report currently being looked at */
-    report: reportPropTypes.isRequired,
-
-    /** The policies which the user has access to and which the report could be tied to */
-    policies: PropTypes.shape({
-        /** Name of the policy */
-        name: PropTypes.string,
-    }),
-
-    /** Route params */
-    route: PropTypes.shape({
-        params: PropTypes.shape({
-            /** Report ID passed via route r/:reportID/details */
-            reportID: PropTypes.string,
-        }),
-    }).isRequired,
+type ReportDetailsPageMenuItem = {
+    key: DeepValueOf<typeof CONST.REPORT_DETAILS_MENU_ITEM>;
+    translationKey: TranslationPaths;
+    icon: FC<SvgProps>;
+    isAnonymousAction: boolean;
+    action: () => void;
+    brickRoadIndicator?: ValueOf<typeof CONST.BRICK_ROAD_INDICATOR_STATUS>;
+    subtitle?: number;
+};
 
+type ReportDetailsPageOnyxProps = {
     /** Personal details of all the users */
-    personalDetails: PropTypes.objectOf(participantPropTypes),
-};
+    personalDetails: OnyxCollection<OnyxTypes.PersonalDetails>;
 
-const defaultProps = {
-    policies: {},
-    personalDetails: {},
+    /** Session info for the currently logged in user. */
+    session: OnyxEntry<OnyxTypes.Session>;
 };
+type ReportDetailsPageProps = {
+    /** The report currently being looked at */
+    report: OnyxEntry<OnyxTypes.Report>;
+} & ReportDetailsPageOnyxProps &
+    WithReportOrNotFoundProps;
```

Comment:
I guess report prop included in `WithReportOrNotFoundProps` type
```suggestion
type ReportDetailsPageProps = ReportDetailsPageOnyxProps & WithReportOrNotFoundProps & StackScreenProps<ReportDetailsNavigatorParamList, typeof SCREENS.REPORT_DETAILS.ROOT>;
```